### ISSUE TITLE
Workflow Step details in ENV

### ISSUE THREAD

**Issue by hekaldama** [rundeck/rundeck#1843]
Would like to suggest adding step details into the ENV such that I could tell what step I am on.

E.g.

I have a job called "Foo" which has a script workflow step(0), a job reference step(1), and a custom plugin step(2). When I inspect the ENV at each of these steps I should see something like:

``` bash
RD_WORKFLOW_STEP_TYPE={script,job reference, custom}
RD_WORKFLOW_STEP_DESCRIPTION={free form user input}
RD_WORKFLOW_STEP_INDEX={0,1,2}
```

I am not sure how to handle labeling error handling steps, and I am open to suggestions.

Let me know if that doesn't make sense.

**Comment by gschueler** [comment]
Good idea, for an error handler could have something like `RD_WORKFLOW_STEP_HANDLER=true`

**Comment by steveblackmon** [comment]
+1

**Comment by fdevans** [comment]
Thanks for the report. As part of a cleanup of our issue tracker, we're closing issues older than 12 months, in line with Rundeck's support policy of one year from a release's date (see the [release calendar](https://docs.rundeck.com/docs/history/release-calendar.html) for currently supported versions). This isn't a judgment that the report is invalid — many of these are likely still real — but we can't verify against a version this old.

If you're still seeing this on a currently supported release (6.1.x or 5.20.x+), please open a new issue using our [bug report template](https://github.com/rundeck/rundeck/issues/new?template=bug_report.yml) with your version and reproduction steps. Please don't reopen this issue — a fresh issue keeps things tracked against a supported version and gets it in front of us properly. We're actively working on improving how issues get submitted and triaged going forward.

Thanks for your patience,
The Rundeck Team
